script.js

```diff
@@ -38,7 +38,29 @@ jQuery(function () {
                     // insert syntax and close dialog
                     success: function (data) {
                         $box.find('.content').addClass('success').text(data.message);
-                        insertAtCarret('wiki__text', '{{:' + data.id + '}}');
+                        if(proseMirrorIsActive)
+                        {
+                            var sel, range;
+                            if (window.getSelection) {
+                                sel = window.getSelection();
+                                if (sel.getRangeAt && sel.rangeCount) {
+                                    range = sel.getRangeAt(0);
+                                    range.deleteContents();
+
+                                    var el = document.createElement("img");
+                                    el.src='/lib/exe/fetch.php?media=' + data.id;
+                                    el.className="media";
+                                    el.draggable="true";
+                                    el.contentEditable="false";
+                                    range.insertNode(el);
+                                }
+                            } else if (document.selection && document.selection.createRange) {
+                                document.selection.createRange().text = '{{:' + data.id + '}}';
```

Comment:
Another question: what is this else branch for? Where would this syntax be inserted?

Reply:
To native DokuWiki editor. Actually it was in original code.

So we defines is it prosemirror and if not just insert markup as it was originally.

Reply:
Oh, now i get it. I`ll look at my production. It wat 2 years ago, really thought this request died already.